workflow/rocoto/workflow_xml.py

```diff
@@ -77,7 +77,10 @@ def _get_workflow_header(self):
         """
 
         scheduler = self._app_config.scheduler
-        cyclethrottle = self._base.get('ROCOTO_CYCLETHROTTLE', 3)
+        if self._app_config.do_efsoi:
+            cyclethrottle = self._base.get('ROCOTO_CYCLETHROTTLE', 7)
```

Comment:
Why is the cycle throttle being changed here for EFSOI?

Reply:
The efsoi task for a given cycle runs when the verifying analysis for the cycle 24h after is present.

Reply:
Okay, that's not really the purpose of the throttle. The task should probably be moved to the cycle where it will actually runs and look backwards to where it needs to. MET+ jobs already do that. We might be able to let it go for now and have it fixed in a follow-up PR. @KateFriedman-NOAA @aerorahul thoughts?

Reply:
I think that's doable

Reply:
I remembered to complication with this. So right now the efsoi task that produces the sensitivities for CDATE waits until the analysis for CDATE+24hrs is available. If it's swapped around so that it runs on CDATE to produce the sensitivities for CDATE-24hrs, then there are 4 efsoi tasks at the beginning of the experiment that don't complete - to calculate the sensitivities for CDATE-24hrs, you need the forecasts from CDATE-24hrs and CDATE-30hrs. One way to approach this is have the task run when the verifying analysis is available, then check if the forecasts exist, and exit and return 0 if not. Are there examples of such behavior? Should be be done at the jjob level?

Reply:
We have several jobs that do not run during the first cycle, so this would be no different except for more cycles. You can add a cycleexist dependency for 24 and 30 hours prior (see jobs like aerosol_init and gldas).

Reply:
It shouldn't. If it does, there are ways around that (we can make the last job "final").

Reply:
I guess I don't understand then what the cycleexistdep provides that the existing current file and task dependencies with cycle_offset values wouldn't. Either way with running the task on the cycle in which the verifying analysis is produced, you have five efsoi tasks in the beginning that never complete (the first cycle and then four regular ones). That's why I suggest doing "stub" instances of the efsoi tasks in the first four complete cycles after the first cycle.

Reply:
Sorry, this might have been confusing: by "first four complete cycles after the first cycle" I mean the cycles where all of the tasks are expected to complete, as opposed to the first cycle.

Reply:
@WalterKolczynski-NOAA @CatherineThomas-NOAA pointed out that gldas does something similar to the stub task instance idea, like this snippet from jobs/JGDAS_ATMOS_GLDAS

```
xtime=$((gldas_spinup_hours+12))
if [ $CDATE -le $($NDATE +$xtime $SDATE) ]; then
    echo "GLDAS needs fluxes as forcing from cycles in previous $xtime hours"
    echo "starting from $SDATE. This gldas cycle is skipped" 
    exit 0
fi
```
I was thinking something similar, but for checking to see if the 24hr/30hr ensemble forecasts exist.

Reply:
@AndrewEichmann-NOAA 
Rather than checking if the forecasts exist, can you check the CDATE/SDATE as it does in gldas?  For EFSOI, this is only meant to catch the first handful of cycles and there are other cases that could trigger a forecast file check, for instance if you're running slowly on WCOSS and the ptmp scrubs.  You don't want later cycles saying the efsoi job completed successfully because of missing files.

Reply:
@CatherineThomas-NOAA Yes, what you suggest makes sense.

Reply:
I'd still prefer this be done at the workflow mesh level so the job doesn't even run (and sit in the queue) when it doesn't run, but this will do for now. If gldas is doing it, that should be changed too. But, we may need to keep it this way for ecflow; not sure.

Reply:
I'm unfamiliar with workflow mesh.

Reply:
At the workflow level (e.g. rocoto).

Reply:
At the workflow level (e.g. rocoto).